### PR TITLE
6161 implement interface common block properties

### LINKED ISSUE #6161: Add interfaces for blocks with common properties
## Description
Many blocks have properties such as facing, on/off etc.

However, it's not currently possible to tell which blocks support e.g. horizontal facing using `instanceof`, despite the actual API implementations being the same for every block, because the property implementations have been done using traits only.

This leads to developers of e.g. world-edit plugins having a really bad time trying to use the API for stuff like rotating blocks.

I've seen various kludgy workarounds for this including `method_exists()` and checking for specific instances, but really we just need to have interfaces to deal with this.

I'd planned to do this even as far back as 4.0.0, but somehow never got around to it.

### PR DESCRIPTION
**Pull request by CoderJoeW**
## Introduction
The goal of this is to give plugins an easier way of accessing common properties of blocks such as facing and on/off

### Relevant issues
Fixes #6161

### CODE REVIEW

**Comment by CoderJoeW** [comment]
@dktapps  Just want to get a quick review of this to make sure its headed in the intended direction

**Comment by cjmustard** [comment]
Would this be better utilized trait, I think the is the direction most items like this follow.

[pocketmine\block\Bell.php#42](https://github.com/pmmp/PocketMine-MP/blob/77db7a8941d3d91c7e452cb3ca469f59fbd8f4df/src/block/Bell.php#L42)

**Comment by CoderJoeW** [comment]
> Would this be better utilized trait, I think the is the direction most items like this follow.
> 
> [pocketmine\block\Bell.php#42](https://github.com/pmmp/PocketMine-MP/blob/77db7a8941d3d91c7e452cb3ca469f59fbd8f4df/src/block/Bell.php#L42)

Probably should use the trait  as there is Horizontal and AnyFacing traits that could be used

Possibly out of scope? Would likely fit more into another PR for cleaning up the code

File: src/block/FacingInterface.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+	/**
+	 * Get the current facing direction of the block
+	 */
+	public function getFacing() : int;
+
+	/**
+	 * Set the current facing direction of the block
+	 */
+	public function setFacing(int $facing) : object;
```
Comment (ShockedPlot7560) [inline src/block/FacingInterface.php:36]: Its block not object

File: src/entity/object/Painting.php [hunk 2]
```diff
@@ -40,7 +41,7 @@
 use pocketmine\world\World;
 use function ceil;
 
-class Painting extends Entity{
+class Painting extends Entity implements FacingInterface{
```
Comment (ShockedPlot7560) [inline src/entity/object/Painting.php:44]: Design error, you're using an interface made for blocks in an entity, I don't think you should do that

**Comment by dktapps** [comment]
Closing due to lack of response to review comments

**Pull request closed without merging**